agent/engine/containermetadata/acquire_metadata.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+
+	"github.com/cihub/seelog"
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// acquireNetworkMetadata parses the docker.NetworkSettings struct and
+// packages the desired metadata for JSON marshaling
+func acquireNetworkMetadata(settings *docker.NetworkSettings) NetworkMetadata {
+	if settings == nil {
+		seelog.Errorf("Failed to acquire network metadata: Network metadata not available or does not exist")
+		return NetworkMetadata{}
+	}
+
+	// This metadata is available in two different places in NetworkSettings
+	ipv4Address := settings.IPAddress
+	ipv4Gateway := settings.Gateway
+	ipv6Address := settings.GlobalIPv6Address
+	ipv6Gateway := settings.IPv6Gateway
+
+	// Network mode is not available for Docker API versions 1.17-1.20
+	networkModeFromContainer := ""
+	if len(settings.Networks) == 1 {
+		for modeFromSettings, containerNetwork := range settings.Networks {
+			networkModeFromContainer = modeFromSettings
+			ipv4Address = containerNetwork.IPAddress
+			ipv4Gateway = containerNetwork.Gateway
+			ipv6Address = containerNetwork.GlobalIPv6Address
```

Comment:
I think it is worthwhile to keep the IPv6 information for the case where the container is in host mode and the host instance has a global IPv6. In addition, empty metadata fields are muted from the output so these fields should do anything for most use cases.